src/background/async.js

```diff
@@ -51,9 +51,24 @@ class AsyncNotifications
 {
   static async create(options) {
     return new Promise((resolve, reject) => {
-      chrome.notifications.create('', options, id => resolve(id));
+      try {
+        chrome.notifications.create('', options, id => resolve(id));        
+      } catch (e) {
+        // This is failing on Firefox as it doesn't support the buttons option for the notification and raises an exception when this is called. (see http://bugzil.la/1190681)
+        // Try again with a subset of options that are more broadly supported
+        chrome.notifications.create('', AsyncNotifications.compatible_options(options), id => resolve(id));
+      }
     });
   }
+
+  static compatible_options(options) {
```

Comment:
Small quibble: I'd rather see this inlined in `AsyncNotifications.create`. This whole async mini-library is meant to be an async/await wrapper around the existing Chrome extension APIs. Where possible, the methods should map directly to the existing Chrome API.

Otherwise, looks good.

Reply:
Sounds good. Will update this today.

Reply:
To be clear here, do you want me to move the code into the `AsynNotifications.create()` method, or should I define a second implementation of the class that would work in Firefox and injected at runtime. I think it's probably the first so I'll do that as it takes two seconds, otherwise let me know what you'd like to see.